### PR TITLE
fix: move clean-tag react dep to peers

### LINKED ISSUE #406: React is a direct dependency of clean-tag
https://github.com/styled-system/styled-system/blob/master/packages/clean-tag/package.json#L16

Hello!

clean-tag has React as a direct dependency any chance this could be moved to a peer dep?

It's really only an issue for us when we're linking modules to be honest, albeit annoying.

Cheers

### PR DESCRIPTION
**Pull request by StevenLangbroek**
fixes #406

### CODE REVIEW

**Comment by StevenLangbroek** [comment]
I can't think of a good reason this would be in deps... @jxnblk?

File: packages/clean-tag/package.json [hunk 1]
```diff
@@ -30,5 +29,8 @@
       "babel-register"
     ],
     "babel": "inherit"
+  },
+  "peerDependencies": {
+    "react": "^0.14.x || ^15.x || ^16.x"
```
Comment (jxnblk) [inline packages/clean-tag/package.json]: I think `React.forwardRef` requires 16.x (can't remember which minor version): https://github.com/styled-system/styled-system/blob/master/packages/clean-tag/index.js#L20

Comment (MatthewLarner) [reply packages/clean-tag/package.json]: Thanks for looking into this!

https://github.com/facebook/react/blob/master/CHANGELOG.md#1630-march-29-2018

16.3 for `forwardRef `

**Comment by jxnblk** [comment]
Closing this out in favor of #433 – this issue can be addressed in the new repo if needed

**Pull request closed without merging**